public-ui/src/app/data-browser/charts/chart-biosex/chart-biosex-react.component.tsx

```diff
@@ -24,13 +26,21 @@ export class BioSexChartReactComponent extends React.Component<Props, State> {
 
   constructor(props) {
     super(props);
-    this.state = {options: null};
+    this.state = {options: null, genderAnalysis: props.genderAnalysis,
+    genderCountAnalysis: props.genderCountAnalysis};
   }
 
   componentDidMount() {
       this.getChartOptions();
   }
 
+  componentDidUpdate(prevProps: Readonly<Props>) {
+    const newAnalysis = this.props.genderAnalysis;
+    if (newAnalysis !== this.state.genderAnalysis) {
+        this.setState({ genderAnalysis: newAnalysis }, this.getChartOptions);
+    }
```

Comment:
Same as suggestion above:
```suggestion
    if (prevProps.genderAnalysis !== this.props.genderAnalysis) {
        this.getChartOptions();
    }
```

Reply:
Thanks, changed this.